test/functional/wallet_spend_unconfirmed.py

```diff
@@ -0,0 +1,321 @@
+        self.log.info("Start test feerate with low feerate unconfirmed input, while subtracting from output")
+        wallet = self.setup_and_fund_wallet("unconfirmed_low_wallet_sffo")
+
+        parent_txid = wallet.sendtoaddress(address=wallet.getnewaddress(), amount=1, fee_rate=1)
+        parent_tx = wallet.gettransaction(txid=parent_txid, verbose=True)
+
+        resulting_fee_rate_funding = self.calc_fee_rate(parent_tx)
+        assert_greater_than(self.target_fee_rate, resulting_fee_rate_funding)
+
+        ancestor_aware_txid = wallet.sendtoaddress(address=self.def_wallet.getnewaddress(), amount=0.5, fee_rate=self.target_fee_rate, subtractfeefromamount=True)
+        ancestor_aware_tx = wallet.gettransaction(txid=ancestor_aware_txid, verbose=True)
+
+        self.assert_spends_only_parent(ancestor_aware_tx, parent_txid)
+
+        resulting_fee_rate = self.calc_fee_rate(ancestor_aware_tx)
+        assert_greater_than_or_equal(resulting_fee_rate, self.target_fee_rate)
+        resulting_ancestry_fee_rate = self.calc_set_fee_rate([parent_tx, ancestor_aware_tx])
+        assert_greater_than_or_equal(resulting_ancestry_fee_rate, self.target_fee_rate)
+        assert_greater_than_or_equal(self.target_fee_rate*1.5, resulting_ancestry_fee_rate)
+
+        self.generate(self.nodes[0], 1)
+        wallet.unloadwallet()
+
+    def run_test(self):
+        self.log.info("Starting UnconfirmedInputTest!")
+        self.target_fee_rate = 30
+        self.def_wallet  = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.generate(self.nodes[0], 110)
+
+        # Test that assumptions about meeting feerate and being able to test it hold
+        self.test_target_feerate_confirmed()
+
+        # Spend unconfirmed input with feerate higher than target feerate
+        self.test_target_feerate_unconfirmed_high()
+
+        # Actual test: Spend unconfirmed input with feerate lower than target feerate. Expect that parent gets bumped to target feerate.
+        self.test_target_feerate_unconfirmed_low()
+
+        # Actual test: Spend unconfirmed input with unconfirmed parent both of which have a feerate lower than target feerate. Expect that both ancestors get bumped to target feerate.
+        self.test_chain_of_unconfirmed_low()
+
+        # Actual test: Spend unconfirmed inputs from two parents with low feerates
+        self.test_two_low_feerate_unconfirmed_parents()
+
+        # Actual test: Spend unconfirmed inputs from two parents with mixed feerates
+        self.test_mixed_feerate_unconfirmed_parents()
+
+        # Actual test: Spend chain with high grandparent low parent
+        self.test_chain_of_high_low()
+
+        # Actual test: Spend chain with low grandparent lower parent
+        self.test_chain_of_high_low_below_target_feerate()
+
+        # Actual test: Check that fee is calculated correctly when bumping while subtracting fee from output
+        self.test_target_feerate_unconfirmed_low_sffo()
```

Comment:
in 337a24e6e4fd77d2fc3bf94e87c2a172c39a8fb0

Missing a test for `bumpfee` RPC?

Reply:
Added!